### ISSUE TITLE
Getting Check failed: mdb_env_open(mdb_env, db_path, 0, 0664) == 0 (112 vs. 0) mdb_env_open failed when using convert_mnist_data.exe

### ISSUE THREAD

**Issue by Coderx7** [Microsoft/caffe#101]
today, when I tried to convert mnist dataset into lmdb, I got the aformentioned error. 
here is the log and how I  tried to run it : 
```
"N:\caffe-master\Build\x64\Release\convert_mnist_data.exe" N:\caffe-master\data\mnist\train-images-idx3-ubyte N:\caffe-master\data\mnist\train-labels-idx1-ubyte N:\caffe-master\examples\mnist\mnist_train_lmdb --backend=lmdb
F0719 15:38:34.150894  7052 convert_mnist_data.cpp:103] Check failed: mdb_env_open(mdb_env, db_path, 0, 0664) == 0 (112 vs. 0) mdb_env_open failed
*** Check failure stack trace: ***
"N:\caffe-master\Build\x64\Release\convert_mnist_data.exe" N:\caffe-master\data\mnist\t10k-images-idx3-ubyte N:\caffe-master\data\mnist\t10k-labels-idx1-ubyte N:\caffe-master\examples\mnist\mnist_test_lmdb --backend=lmdb
F0719 15:45:54.102185 14160 convert_mnist_data.cpp:103] Check failed: mdb_env_open(mdb_env, db_path, 0, 0664) == 0 (112 vs. 0) mdb_env_open failed
*** Check failure stack trace: ***
```

what is wrong here, Am I missing something, or is this a bug concerning either lmdb or convert_mnist_dataset?

**Comment by Coderx7** [comment]
when using Leveldb, there is no problem. this happens only when lmdb is used as backend.

**Comment by mnidza** [comment]
@Coderx7 Is it possible that the LMDB folders that you want to create already exist? If so, then try deleting them.

**Comment by Coderx7** [comment]
@mnidza , no, it is not.I checked them.,I know thats not the case, I am 100% sure thats a problem with lmdb itself. 
on windows, the lmdb size is hardcoded as 100 Mb, this causes creating bigger datasets taking more than 100 MB to crash, and one needs to increase the size. interestingly if someone sets the size higher  (even when the space is available), chances are he gets the errors again. 
I guess a more reasonable thing to do is to use leveldb as the default for examples running in caffe instead of lmdb. this way both windows and linux would match and the hassles of getting this to work on different platforms would be minimized.

**Comment by mnidza** [comment]
@Coderx7 I think LMDB map size is no longer hardcoded (see [this commit](https://github.com/Microsoft/caffe/commit/90426645c36ad71c778c4ac3688ec164242a50a1)). Are you using the latest master? I'm guessing not, because line 103 of current [`convert_mnist_data.cpp`](https://github.com/Microsoft/caffe/blob/master/examples/mnist/convert_mnist_data.cpp) does not check any condition.

Also, the error that you are getting does not seem consistent with LMDB map being full. It seems to be a problem with creating/accessing the LMDB folder at the very start, before any data is actually added to it. Did you check that you have write access to the LMDB folder?

**Comment by Coderx7** [comment]
Oh I see, yes, I'm not using the latest version!
Thanks for the heads up. 
and for the second part, no thats not the permission related error. since when I changed it to leveldb, it runs just fine!.
